lib/json_rpc.dart

```diff
@@ -78,8 +78,9 @@ class RPCResponse {
 
 /// Exception thrown when an the server returns an error code to an rpc request.
 class RPCError implements Exception {
-  const RPCError(this.errorCode, this.message, this.data);
+  const RPCError(this.errorCode, this.message, this.data, [this.id]);
 
+  final int? id;
```

Comment:
i added rpc id to the error response since we need to identifiy which request had the error. this is a must and doesnt bother with the original API